src/main/java/edu/harvard/iq/dataverse/marketplace/controller/api/AuthController.java

```diff
@@ -0,0 +1,260 @@
+            UserDetailsImpl loggedInUser = (UserDetailsImpl) authentication.getPrincipal();
+            Role role = roleRepository.findById(roleId).orElse(null);
+
+            if (loggedInUser.getId().equals(userId) && role.getName().equals("ADMIN")) {
+                return ResponseEntity.badRequest().body(
+                    new ServerMessageResponse(HttpStatus.BAD_REQUEST,
+                        "Cannot remove role ADMIN from self.",
+                        "Error during the removal of the role ADMIN, a user cannot remove a role ADMIN from themselves.")
+                );
+            }
+            
+            User user = userRepository.findById(userId).orElse(null);
+
+            if(!user.getRoles().contains(role)) {
+                return ResponseEntity.badRequest().body(
+                    new ServerMessageResponse(HttpStatus.BAD_REQUEST,
+                        "Role not assigned.",
+                        "Error during the removal of the role, the user does not have this role.")
+                );
+            }
+            user.getRoles().remove(role);
+            userRepository.save(user);
+
+            ServerMessageResponse messageResponse = new ServerMessageResponse(HttpStatus.OK,
+                    "User role was successfully removed.",
+                    String.format("The user %s was successfully removed the role %s.", user.getUsername(), role.getName()));
+            
+            return ResponseEntity.ok(messageResponse);
+            
+        } catch (Exception e) {
+            return ResponseEntity.internalServerError().body(
+                new ServerMessageResponse(HttpStatus.INTERNAL_SERVER_ERROR,
+                    e.getClass().getName(),
+                    e.getStackTrace()[0].toString())
+            );
+        }
+    }
+
+    @PutMapping("/password/update")
+    @AuthAPIDocs.ChangePassword
+    public ResponseEntity<?> changePassword(@RequestHeader("password") String password) {
+        
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        UserDetailsImpl loggedInUser = (UserDetailsImpl) authentication.getPrincipal();
+        User user = userRepository.findById(loggedInUser.getId()).orElse(null);
+        user.setPassword(encoder.encode(password));
+        userRepository.save(user);
+
+        return ResponseEntity.ok(new ServerMessageResponse(HttpStatus.OK,
+                "Password updated successfully.",
+                "The password was successfully updated."));
+    }
+    
+    @GetMapping("/roles")
+    @PreAuthorize("hasAuthority('ADMIN')")
+    @AuthAPIDocs.GetRoles
+    public ResponseEntity<?> getRoles() {
+
+        Set<RoleDTO> rolesDTO = new HashSet<>();
+        List<Role> roles = roleRepository.findAll();
+
+        for (Role role : roles) {
+            rolesDTO.add(new RoleDTO(role));
+        }
+
+        return ResponseEntity.ok(rolesDTO);
+    }
+
+    
```

Comment:
As discussed two extra endpoints might be nice:

- list all users
- list user by id